lfs/gitscanner_catfilebatch.go

```diff
@@ -23,42 +23,41 @@ func runCatFileBatch(pointerCh chan *WrappedPointer, revs *StringChannelWrapper,
 		return err
 	}
 
-	go catFileBatchOutput(pointerCh, cmd, errCh)
-	go catFileBatchInput(cmd, revs, errCh)
-	return nil
-}
+	go func() {
+		scanner := &catFileBatchScanner{r: cmd.Stdout}
+		for r := range revs.Results {
+			cmd.Stdin.Write([]byte(r + "\n"))
+			hasNext := scanner.Scan()
```

Comment:
OK, after some meat-space conversation, I get this now 👍 . `hasNext` returns whether or not we can scan again, and we may or _may not_ get a pointer, or an error. I think renaming this to `canScanAgain` and documenting the `Scan()` method itself would be ✨